File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.py
t = int(input())
nums = [int(input()) for _ in range(t)]

fibo = []
fibo.append([1, 0])
fibo.append([0, 1])
for i in range(2, max(nums) + 1):
    z = fibo[i - 1][0] + fibo[i - 2][0]
    o = fibo[i - 1][1] + fibo[i - 2][1]
    fibo.append([z, o])

for n in nums:
    print(*fibo[n])

File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.js
const fs = require('fs'); 
const [T, ...Ns] = fs.readFileSync(0, 'utf-8').toString().trim().split('\n').map(Number);

const fibo = []
fibo[0] = [1, 0];
fibo[1] = [0, 1];
for(let n = 2; n <= Math.max(...Ns); n++){
    const z = fibo[n - 1][0] + fibo[n - 2][0]
    const o = fibo[n - 1][1] + fibo[n - 2][1]
    fibo.push([z, o])
}

for(const N of Ns){
    console.log(fibo[N].join(' '))
}